### DISCUSSION TITLE
✨ allow the binding adress to be set when tunneling a service

### DISCUSSION THREAD

**Discussion by borissmidt** [dagger/dagger#12482] [Ideas]
### What are you trying to do?

Currently services tunneled to the host are exposed as *:<port>.

It would be easoer to avoid port collision if we could expose the service on a certain adress on the host. 
I.e. 127.0.0.2 for example.

```go
client.Host().Tunnel(httpSrv, {Native:true, Host: "127.0.0.2").Start(ctx)
```

### Why is this important to you?

he reason for this feature is that during the development phase and when debugging unit tests it is easier to run the service directly in your ide.

Often there are services on localhost 127.0.0.1 but 127.0.0.2 and up are almost always available.

To have a consistent environment between the local machine and the dagger environment can be done by setting the services in /etc/hosts

```
127.0.0.2 mysql
127.0.0.2 mongodb
127.0.0.3 toxiproxy
```

### How are you currently working around this?

Manually managing the host ports since we don't want to update the tests with the random ports.

**Comment by marcosnils** [comment]
I personally have conflicting thoughts about supporting this. 

1 - I haven't seen and/or used this anywhere else (docker, k8s, podman, etc). I'm also unsure if all other OSs have mapped 127.0.0.1/8  to the same interface
2 - Same thing can be achieved by mapping ports to different ports in the host and then configuring the app appropriately to use different ports for such purpose.

**Comment by borissmidt** [comment]
I think the use of the 127.0.0.0/8 is just not used often since there isn't a convenient 'localhost2' available by default on the systems. 

After a short research most of the tools mentioned  support the use of other addresses than 127.0.0.1.
```
podman run -p 127.0.0.4:3306:3306  -e MYSQL_ROOT_PASSWORD=my-secret-pw docker.io/mysql:8.0.35-debian
docker run -p 127.0.0.3:3306:3306  -e MYSQL_ROOT_PASSWORD=my-secret-pw mysql:8.0.35-debian
kubectl port-forwards --address 127.0.0.5  pods/mysql-123 3306:3306
```

It seems that only mac has an issue with using anything other than 127.0.0.1 by default.
https://superuser.com/questions/458875/how-do-you-get-loopback-addresses-other-than-127-0-0-1-to-work-on-os-x

But windows and linux do have support out of the box.

**Comment by marcosnils** [comment]
> After a short research most of the tools mentioned support the use of other addresses than 127.0.0.1.

This is a good enough examples to support it IMO. If other tools support it, I don't see why we wouldn't also. Thx for sharing @borissmidt
